plugins/policy-recommendation/policy_recommendation_job_test.py

```diff
@@ -0,0 +1,1215 @@
+                        cidr = "192.168.0.1/32",
+                    )
+                )
+            ],
+            ports=[
+                kubernetes.client.V1NetworkPolicyPort(
+                    port = 80,
+                    protocol = "TCP"
+                )
+            ]
+        )
+    ),
+    (
+        'fd12:3456:789a:1::1#80#TCP',
+        kubernetes.client.V1NetworkPolicyEgressRule(
+            to=[
+                kubernetes.client.V1NetworkPolicyPeer(
+                    ip_block = kubernetes.client.V1IPBlock(
+                        cidr = "fd12:3456:789a:1::1/128",
+                    )
+                )
+            ],
+            ports=[
+                kubernetes.client.V1NetworkPolicyPort(
+                    port = 80,
+                    protocol = "TCP"
+                )
+            ]
+        )
+    ),
+])
+def test_generate_k8s_egress_rule(test_input, expected_egress_rule):
+    if test_input == "wrongformat":
+        with pytest.raises(SystemExit) as pytest_wrapped_e:
+                generate_k8s_egress_rule(test_input)
+        assert pytest_wrapped_e.type == SystemExit
+        assert pytest_wrapped_e.value.code == 1
+    else:
+        egress_rule = generate_k8s_egress_rule(test_input)
+        assert egress_rule == expected_egress_rule
+
+@pytest.mark.parametrize("test_input, expected_ingress_rule", [
+    (
+        'wrongformat',
+        "",
+    ),
+    (
+        'antrea-test#{"podname":"perftest-a"}#5201#TCP',
+        kubernetes.client.V1NetworkPolicyIngressRule(
+            _from=[
+                kubernetes.client.V1NetworkPolicyPeer(
+                    namespace_selector = kubernetes.client.V1LabelSelector(
+                        match_labels = {
+                            "name":'antrea-test'
+                        }
+                    ),
+                    pod_selector = kubernetes.client.V1LabelSelector(
+                        match_labels = {
+                            "podname":"perftest-a"
+                        }
+                    ),    
+                )
+            ],
+            ports=[
+                kubernetes.client.V1NetworkPolicyPort(
+                    port = 5201,
+                    protocol = "TCP"
+                )
+            ]
+        )
+    ),
+])
+def test_generate_k8s_ingress_rule(test_input, expected_ingress_rule):
+    if test_input == "wrongformat":
+        with pytest.raises(SystemExit) as pytest_wrapped_e:
+                generate_k8s_ingress_rule(test_input)
+        assert pytest_wrapped_e.type == SystemExit
+        assert pytest_wrapped_e.value.code == 1
+    else:
+        ingress_rule = generate_k8s_ingress_rule(test_input)
+        assert ingress_rule == expected_ingress_rule
+
+@pytest.mark.parametrize("test_input, expected_policy_name", [
+    (
+        "recommend-k8s-np",
+        "recommend-k8s-np-y0cq6"
+    ),
+    (
+        "recommend-allow-anp",
+        "recommend-allow-anp-y0cq6"
```

Comment:
I admit I don't understand how can guarantee a specific suffix is appended.

Reply:
By calling `random.seed(0)`, we could make sure the randomly generated suffix are the same.

Reply:
Got it - so you're relying on seeding the number generator in the same way so that it always generates the same sequence. That should work as long as we run the tests on linux/amd64, as far as I recall